test/core/image_processing/anisotropic_diffusion.cpp

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2020 Olzhas Zhumabek <anonymous.from.applecity@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_processing/diffusion.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/pixel.hpp>
+#include <boost/gil/typedefs.hpp>
+
+#include <boost/core/lightweight_test.hpp>
+
+#include <limits>
+#include <random>
+
+namespace gil = boost::gil;
+
+template <typename ImageType>
+void heat_conservation_test(std::uint32_t seed)
+{
+    std::mt19937 twister(seed);
+    std::uniform_int_distribution<gil::uint8_t> dist;
```

Comment:
If you add
```cpp
#include "core/test_fixture.hpp"
```
then you should be able to use our facade on those two:
```cpp
gil::test::fixture::random_value<std::uint8_t> random;
```
If you also add this extra constructor to the `random_value`
```cpp
 random_value(std::uint32_t seed) : rng_(seed), uid_(range_min, range_max) {}
```
then, obviously, you can use
```cpp
gil::test::fixture::random_value<std::uint8_t> random(seed);
```

Reply:
I believe we could also change default constructor to not use `std::random_device`, as not being able to reproduce the results is not useful. It should not break any test if the stuff they are testing is correct. What do you think?

Reply:
auto value = uid_(rng_);
    BOOST_ASSERT(range_min <= value && value <= range_max);
This assert looks useless too, as `value` comes from a distribution of `[range_min, range_max]`.

Reply:
I have added a few changes to `random_value`, please have a look

Reply:
@simmplecoder Your changes there look good to me. Thanks